### PR TITLE
JAMES-3694 Add a TTL on RabbitMQ queues

### PR DESCRIPTION
**Pull request by vttranlina**
Jira: https://issues.apache.org/jira/browse/JAMES-3694

### CODE REVIEW

File: backends-common/rabbitmq/src/main/java/org/apache/james/backends/rabbitmq/RabbitMQConfiguration.java [hunk 1]
```diff
@@ -488,6 +492,12 @@ public Builder hosts(Collection<Host> hosts) {
             return this;
         }
 
+        public Builder queueTTL(long ttl) {
+            Preconditions.checkArgument(ttl > 0, String.format("'%s' must be strictly positive", QUEUE_TTL));
```
Comment (chibenwa) [inline backends-common/rabbitmq/src/main/java/org/apache/james/backends/rabbitmq/RabbitMQConfiguration.java:496]:
```suggestion
            Preconditions.checkArgument(ttl > 0, "'%s' must be strictly positive", QUEUE_TTL);
```

String formatting is built in checkArgument

File: server/apps/distributed-app/docs/modules/ROOT/pages/configure/rabbitmq.adoc [hunk 2]
```diff
@@ -129,6 +129,10 @@ It can be useful then to disable it.
 Tips for choosing metricsEnabled are explained in
 https://github.com/apache/james-project/blob/master/server/apps/distributed-app/sample-configuration/rabbitmq.properties[rabbitmq.properties]
 
+| notification.queue.ttl
+| Configure queue ttl (in ms). References: https://www.rabbitmq.com/ttl.html#queue-ttl.
+Optional integer, defaults is 3600000.
```
[the remark this answers is not in the document]
Comment (chibenwa) [reply server/apps/distributed-app/docs/modules/ROOT/pages/configure/rabbitmq.adoc:134]: Yes IMO this shoud, be optional, if ommitted we should NOT add the TTL.

File: backends-common/rabbitmq/src/main/java/org/apache/james/backends/rabbitmq/QueueArguments.java [hunk 3]
```diff
@@ -31,7 +31,7 @@ public Builder deadLetter(String deadLetterQueueName) {
         }
 
         public Builder queueTTL(long queueTTL) {
-            arguments.put("x-expires", queueTTL);
+            arguments.put("x-message-ttl", queueTTL);
```
Comment (chibenwa) [inline backends-common/rabbitmq/src/main/java/org/apache/james/backends/rabbitmq/QueueArguments.java:34]: No it is not the same thing...

File: event-bus/distributed/src/test/java/org/apache/james/events/RabbitMQEventBusTest.java [hunk 4]
```diff
@@ -414,6 +415,34 @@ void dispatchShouldPublishSerializedEventToRabbitMQWhenNotBlocking() {
...
+                .block();
+
+            eventBus.dispatch(EVENT, NO_KEYS).block();
+
+            byte[] messageBody = rabbitMQExtension.getReceiverProvider().createReceiver().consumeAutoAck(ttlQueue)
+                .blockFirst()
+                .getBody();
+
+            assertThat(eventSerializer.asEvent(new String(messageBody, StandardCharsets.UTF_8))).isEqualTo(EVENT);
+        }
```
Comment (chibenwa) [inline event-bus/distributed/src/test/java/org/apache/james/events/RabbitMQEventBusTest.java:444]: Can you explain the rationals behind this test?

File: backends-common/rabbitmq/src/main/java/org/apache/james/backends/rabbitmq/QueueArguments.java (hunk 3 above)
Comment (Arsnael) [reply backends-common/rabbitmq/src/main/java/org/apache/james/backends/rabbitmq/QueueArguments.java:34]: I think you misunderstood here. We want to put a TTL on the queue, not on the messages in the queue :) Thus your added test at the end does not make sense as well in the context that we want

Comment (vttranlina) [reply backends-common/rabbitmq/src/main/java/org/apache/james/backends/rabbitmq/QueueArguments.java:34]: you are right, I re-read the ticket, I  misunderstood here

**Review by Arsnael — APPROVED** [review]

File: server/apps/distributed-app/sample-configuration/rabbitmq.properties [hunk 5]
```diff
@@ -73,3 +73,8 @@ mailqueue.size.metricsEnabled=false
 # Disable with caution (this only makes sense in a distributed setup where other nodes consume tasks).
 # Defaults to true.
 task.consumption.enabled=true
+
+# Configure queue ttl (in ms). References: https://www.rabbitmq.com/ttl.html#queue-ttl.
+# This is used only on queues used to share notification patterns, are exclusive to a node. If omitted, it will not add the TTL configure when declaring queues.
+# Optional integer, defaults is 3600000.
+notification.queue.ttl=3600000
```
Comment (chibenwa) [inline server/apps/distributed-app/sample-configuration/rabbitmq.properties:80]:
```suggestion
#notification.queue.ttl=3600000
```

Let's not activate this by default

**Review by chibenwa — APPROVED** [review]

**Review by chibenwa — APPROVED** [review]

**Review by Arsnael — APPROVED** [review]

**Pull request merged**